src/object/cli_shard.c

```diff
@@ -1109,15 +1109,6 @@ dc_enumerate_cb(tse_task_t *task, void *arg)
 			D_DEBUG(DB_TRACE, "rpc %p RPC %d may need retry: "
 				""DF_RC"\n", enum_args->rpc, opc, DP_RC(rc));
 		} else {
-			if (enum_args->eaa_obj) {
-				struct dc_obj_shard *shard;
-
-				shard = enum_args->eaa_obj;
-				shard->do_pl_shard.po_target = -1;
-				D_ERROR(DF_OID" set shard %d invalid.\n",
-					DP_OID(shard->do_id.id_pub),
-					shard->do_pl_shard.po_shard);
-			}
```

Comment:
why remove it?

Reply:
oh, this suppose to help shard choosing( exclude the failure one), but it will cause some test failures, if the test will only inject ONCE failure on the server side, i.e. server only fail once, then go back normal.  In this case, the shard should be keep healthy. So I removed this, and update shard status only when pool map is refreshed.